src/components/fx/calc.js

```diff
@@ -10,9 +10,33 @@
 
 var Lib = require('../../lib');
 var Registry = require('../../registry');
+var baseAttrs = require('../../plots/attributes');
 
 module.exports = function calc(gd) {
     var calcdata = gd.calcdata;
+    var fullLayout = gd._fullLayout;
+
+    function makeCoerceHoverInfo(fullTrace) {
+        var moduleAttrs = fullTrace._module.attributes;
+        var attrs = moduleAttrs.hoverinfo ?
+            {hoverinfo: moduleAttrs.hoverinfo} :
+            baseAttrs;
+        var valObj = attrs.hoverinfo;
+        var dflt;
+
+        if(fullLayout._dataLength === 1) {
+            var flags = valObj.dflt === 'all' ?
+                valObj.flags.slice() :
+                valObj.dflt.split('+');
+
+            flags.splice(flags.indexOf('name'), 1);
+            dflt = flags.join('+');
+        }
+
+        return function(val) {
+            return Lib.coerce({hoverinfo: val}, {}, attrs, 'hoverinfo', dflt);
```

Comment:
This here coerces `hoverinfo` items with no valid flags to `'all'` if `gd.data.length > 1` or all flags but `'name'` on single-trace graphs. For example,

```js
hoveinfo: ['sda', null, 'hello+sup']

// assuming more than one trace,
// => gets coerced to
hoverinfo: ['all', 'all', 'all']
```

---

This is debatable I feel. Perhaps, `hoverinfo` items with no valid flags should get coerced to `'none'` instead? This would be more in line with e.g. `marker.color` where

```js
marker.color: ['red', null, 'blue']

// gets coerced to
marker.color: ['red', '#444' /* i.e. Color.defaultLine */, 'blue']
```

I'm currently leaning toward the second option, but I have no strong opinion.